names/names.py

```diff
@@ -10,13 +10,30 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+## Original double for loop garbage
+
+# duplicates = []
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+## Solution using sets
+
+name1 = set(names_1)
+name2 = set(names_2)
+duplicatesB = name1 & name2
```

Comment:
Did you time this? Look up https://wiki.python.org/moin/TimeComplexity and look at `intersection`. If you look at the time complexity of intersection you will see that it is a `little` bit better than O(n^2). So even though it is a LOT less code. It would be almost the same runtime. But I could be wrong. Double check and let me know.

That being said you did have the right idea by turning into a `set`. That gives us O(1) lookup time. Knowing that, you could do something like 
```
duplicates = []
names_2_set = set(names_2)
for name in names_1:
    if name in names_2_set:
      duplicates.append(name)
```